pkg/testworkflows/testworkflowprocessor/processor.go

```diff
@@ -356,6 +366,9 @@ func (p *processor) Bundle(ctx context.Context, workflow *testworkflowsv1.TestWo
 			if !filepath.IsAbs(containers[i].VolumeMounts[j].MountPath) {
 				containers[i].VolumeMounts[j].MountPath = filepath.Clean(filepath.Join(workingDir, containers[i].VolumeMounts[j].MountPath))
 			}
+			if _, ok := volumeNameMap[containers[i].VolumeMounts[j].Name]; ok {
```

Comment:
The mounted volumes will differ per container, so there should be container or step granularity. Otherwise, you may use directory that is not mounted and treat it as secret volume:

```
steps:
  pod:
    volumes:
    - name: secret-vol
      secret: { secretName: "some-secret" }
- container:
     volumeMounts:
     - mountPath: /usr/bin
        name: "secret-vol"
  shell: cat /usr/bin/something
- shell: "echo this step will take looooong as image's /usr/bin will be treat as secret"
```

Reply:
looks like an edge case, but agree, need container granularity